flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/FlinkConfMountDecorator.java

```diff
@@ -148,6 +149,7 @@ private Pod decoratePod(Pod pod) {
 
 		// remove kubernetes.config.file
 		propertiesMap.remove(KubernetesConfigOptions.KUBE_CONFIG_FILE.key());
+		propertiesMap.remove(DeploymentOptionsInternal.CONF_DIR.key());
```

Comment:
I think it would be safer to use `flinkConfig.removeConfig()` which also removes fallback keys.

Reply:
We could clone a flink config and use `flinkConfig.removeConfig()` here. I will fix it.